### ISSUE TITLE
[BUG] DigestAuth with SetBody will be 'http: ContentLength=54 with Body length 0'

### ISSUE THREAD

**Issue by icepie** [go-resty/resty#729]
### Code

```go
	req := c.HttpClient.R().
		SetDigestAuth("admin", "123456").
		SetHeader("Content-Type", "application/xml").
		SetBody(reqBody)

	resp, err := req.Put("/ISAPI/AccessControl/RemoteControL/door/1")

```

### Out

```
2023/10/11 17:13:52  Put "http://192.168.0.51/ISAPI/AccessControl/RemoteControL/door/1": net/http: HTTP/1.x transport connection broken: http: ContentLength=54 with Body length 0
```

but `GET` method will be fine !

**Comment by SVilgelm** [comment]
there is a special function to automatically set the content length, it will also set to 0 if needed:

```go
r.SetContentLength(true)
```

**Comment by icepie** [comment]
> there is a special function to automatically set the content length, it will also set to 0 if needed:
> 
> ```go
> r.SetContentLength(true)
> ```

Code
```go

	req := c.HttpClient.R().SetDigestAuth(c.username, c.password).
		SetHeader("Content-Type", "application/xml").
		SetContentLength(true).
		SetBody(reqBody)

	resp, err := req.
		Put("/ISAPI/AccessControl/RemoteControL/door/1")

```


Out

```
 Put "http://192.168.0.51/ISAPI/AccessControl/RemoteControL/door/1": net/http: HTTP/1.x transport connection broken: http: ContentLength=54 with Body length 0
```

Same problem.

**Comment by jeevatkm** [comment]
Merged #730
